src/ioctls/vm.rs

```diff
@@ -1087,8 +1088,21 @@ impl VmFd {
     ///     fd: 0,
     ///     flags: KVM_CREATE_DEVICE_TEST,
     /// };
-    /// let device_fd = vm
-    ///     .create_device(&mut device).unwrap();
+    /// // On ARM, creating VGICv3 may fail due to hardware dependency.
+    /// // Retry to create VGICv2 in that case.
+    /// let device_fd = match vm.create_device(&mut device) {
+    ///     Ok(fd) => fd,
+    ///     Err(_) => {
+    ///         #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    ///         panic!("Cannot create VFIO device.");
+    ///         #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
+    ///         {
+    ///             device.type_ = kvm_device_type_KVM_DEV_TYPE_ARM_VGIC_V2;
+    ///             vm.create_device(&mut device)
+    ///                 .expect("Cannot create vGIC device")
+    ///         }
+    ///     }
+    /// };
```

Comment:
```suggestion
    /// let device_fd = vm.create_device(&mut device).unwrap_or_else(|_| {
    ///     #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
    ///     panic!("Cannot create VFIO device.");
    ///     #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
    ///     {
    ///         device.type_ = kvm_device_type_KVM_DEV_TYPE_ARM_VGIC_V2;
    ///         vm.create_device(&mut device).expect("Cannot create vGIC device")
    ///     }
    /// });
```

Reply:
Thanks. 
The suggestion in multi-lines doesn't apply online. I edited locally and updated.